### PR TITLE
Add policy for build-time dependency vulns

### PR DESCRIPTION
**Pull request by ionut-arm**
Any and all feedback is welcome :)

Related to #855

### CODE REVIEW

**Comment by PushkarJ** [comment]
Thank you @ionut-arm while folks are reviewing the content, can you take a look at Lint and spelling failures and try to fix them!

**Comment by PushkarJ** [comment]
Thank you @ionut-arm while folks are reviewing the content, can you take a look at Lint and spelling failures and try to fix them!

File: policy/overview-policy-build-time-dependency-vulns.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+### Identifying vulnerabilities in your build-time dependencies
+
+Checks for vulnerabilities in your project should happen automatically and
+continuously.
+
+Tooling around this issue typically piggybacks on a build system, such as
+[`npm-audit`](https://docs.npmjs.com/cli/v8/commands/npm-audit) or
+[`cargo-audit`](https://docs.rs/cargo-audit/latest/cargo_audit/index.html),
+however more generic solutions exist (e.g., the [Github
```
Comment (stmcginnis) [inline policy/overview-policy-build-time-dependency-vulns.md:57]: Very trivial nit, but would be good to use the proper casing for GitHub.

**Review by stmcginnis — COMMENTED** [review]
Looks great. This covers a few aspects of dependency vulnerability handling that may not be obvious to all folks, so really nice to get this written out and added.

File: policy/overview-policy-build-time-dependency-vulns.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+1. As a developer of a library, how do you advertise the discovery of a
+   vulnerability to your dependents?
+2. As a consumer of other libraries, how can you tell if one of your
+   dependencies is affected by a vulnerability?
+3. As a consumer of other libraries, how do you mitigate the effect of a
+   vulnerability in a dependency? Does this change with how far up the
+   dependency tree the issue is? How do you advertise the discovery of such
+   vulnerabilities?
+
```
Comment (jpmcb) [inline policy/overview-policy-build-time-dependency-vulns.md:19]: Would it make sense to also make recommendations on "As a developer of a library, how to recommend users report new vulnerabilities in your library"?

Usually, a dedicated "reporting" mechanism (a security emailing list, GitHub's integrated tools to securely report a vulnerability, etc.) should be used vs. a user opening an issue or making some public declaration of their new finding.

File: policy/overview-policy-build-time-dependency-vulns.md (hunk 1 above)
Comment (ionut-arm) [reply policy/overview-policy-build-time-dependency-vulns.md]: Thanks for spotting it, fixed!

File: policy/overview-policy-build-time-dependency-vulns.md (hunk 2 above)
Comment (ionut-arm) [reply policy/overview-policy-build-time-dependency-vulns.md:21]: Added! Let me know if it roughly aligns with what you'd expect.

**Comment by PushkarJ** [comment]
Brought this up in TAG Security check-in meeting today. Will keep it open for a week for further feedback. We could also write a blog post on this topic. Maybe @ashutosh-narkar you can help here?

**Comment by nadgowdas** [comment]
Looks great ! There's also a vulnerability disclosure process for open source projects being documented here: https://github.com/ossf/oss-vulnerability-guide/blob/main/guide.md 
I think it might be relevant in this context.

**Comment by ionut-arm** [comment]
> Looks great ! There's also a vulnerability disclosure process for open source projects being documented here: https://github.com/ossf/oss-vulnerability-guide/blob/main/guide.md I think it might be relevant in this context.

Thanks, I've added a couple of links to that guide as well!

File: policy/overview-policy-build-time-dependency-vulns.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+vulnerability disclosure for open source
+software](https://github.com/ossf/oss-vulnerability-guide) created by the
+OpenSSF.
+
+### Disclosing vulnerabilities in your library
+
+Disclosure of vulnerabilities should generally be done in a standardized fashion
+to allow downstream users to identify issues automatically regardless of how far
+the library is on the dependency chain. The [OpenSSF
+guide](policy/overview-policy-build-time-dependency-vulns.md) mentioned
```
Comment (PushkarJ) [inline policy/overview-policy-build-time-dependency-vulns.md:61]: Not sure if this hyperlink is needed. It is also failing spellcheck and lint, fwiw.

**Review by PushkarJ — APPROVED** [review]
Thank you so much for adding this to our growing list of guidances

**Pull request merged**